### ISSUE TITLE
Could you please check that the way you read labels gives wrong labels?

### ISSUE THREAD

**Issue by minhmanho** [kazuto1011/deeplab-pytorch#56]
As coded in cocostuff.py, you read the the semantic png map as gray-scale.
https://github.com/kazuto1011/deeplab-pytorch/blob/6755e22153b0db96e11b84183efb39c17be72191/libs/datasets/cocostuff.py#L83

It's convenient but actually this way give labels from 0 -> 181, while the original labels contains 182 (including unlabeled) (ref: https://github.com/nightrome/cocostuff/blob/master/labels.txt).
I also double-checked the images corresponding to the labels this way give as follows:
Image for label 1, yours is person but original is bicyble:
![000000224051](https://user-images.githubusercontent.com/24385510/54864274-5cdf0c00-4d98-11e9-8842-4de5acb1cc4c.jpg)

Image for label 23, yours is bear but original is zebra:
![000000031269](https://user-images.githubusercontent.com/24385510/54864296-c19a6680-4d98-11e9-9133-6fe9f2299660.jpg)

I understand your way still works but should we add +1 for the labels?

**Comment by kazuto1011** [comment]
Although the official labels.txt list 0 as the "unlabeled" class,  the actual png images contain it as 255 like PASCAL VOC; the labels contain 0-181 and 255. The previous annotation files in 10k subset were compatible with the 0-182 labels while the ones in 164k are not. Please check [the author's note](https://github.com/nightrome/cocostuff#downloads) in the 164k page.
> **Caffe-compatible stuff-thing maps** We suggest using the stuffthingmaps, as they provide all stuff and thing labels in a single .png file per image. Note that the .png files are indexed images, which means they store only the label indices and are typically displayed as grayscale images. To be compatible with most Caffe-based semantic segmentation methods, thing+stuff labels cover indices 0-181 and 255 indicates the 'unlabeled' or void class.

Moreover, `torch.nn.CrossEntropyLoss` expects that label indices to be used in loss calculation range from 0. To exclude unlabeled regions from the loss, I recommend using the current numbering as is and referring to the mapping provided in this repository (just shifted by 1 and deleted unlabeled).
https://github.com/kazuto1011/deeplab-pytorch/blob/master/data/datasets/cocostuff/labels.txt

**Comment by minhmanho** [comment]
Thank you so much for your clarification.
